youtube_dl/extractor/common.py

```diff
@@ -438,6 +439,10 @@ def _guess_encoding_from_content(content_type, webpage_bytes):
     def _webpage_read_content(self, urlh, url_or_request, video_id, note=None, errnote=None, fatal=True, prefix=None, encoding=None):
         content_type = urlh.headers.get('Content-Type', '')
         webpage_bytes = urlh.read()
+        webpage_bytes, bom_enc = parse_strip_bom(webpage_bytes)
```

Comment:
1. Explicit `encoding` may be added to `_parse_json` and forwarded there from `_download_json`. With this one will be able to specify encoding manually, `utf-8-sig` in this case.
2. Drop BOM in `_parse_json` leaving webpage intact.